### PR TITLE
#162948806 Delete a Posted Article

### PR DESCRIPTION
**Pull request by KMaina**
#### What does this PR do?
Allows for an authenticated user to delete an article
#### Description of Task to be completed?
Delete an article that has been posted by a user.
#### How should this be manually tested?
- Clone the repo locally.
- Checkout to the `ft-delete-article-162948806 branch`.
- Run `npm install` to install all dependencies.
- Navigate to the `/login` page to login successfully.
- Navigate to the `/article/:article` page and click on the delete button.
#### Any background context you want to provide?
- I might have some errors in the `ArticleDetail` page since I rebased off a branch that was yet to be merged. Hopefully, their work will be successfully merged and I can rebase to get the right changes.
- You can only delete an article you created.
#### What are the relevant pivotal tracker stories?
[#16294880](https://www.pivotaltracker.com/story/show/162948806)

### CODE REVIEW

File: src/pages/ArticleDetail/index.jsx [hunk 1]
```diff
@@ -1,53 +1,64 @@
...
+
+export const mapStateToProps = state => {
+  return state.singleArticleReducer;
 };
 
-export default ArticleDetail;
+export const mapDispatchToProps = dispatch => ({
+  getSingleArticleDispatch: article =>
+    dispatch(getSingleArticleDispatch(article)),
+  deleteArticle: (article, st) => dispatch(deleteArticle(article, st))
```
Comment (d-kahara) [inline src/pages/ArticleDetail/index.jsx]: Consider renaming `st` to something a bit more descriptive

**Comment by Georgeygigz** [comment]
Good work @KMaina, Kindly rebase your work

**Comment by Georgeygigz** [comment]
Good work @KMaina, kindly rebase your work

**Comment by andrewhingah** [comment]
Looks good

File: src/pages/ArticleDetail/index.jsx [hunk 2]
```diff
@@ -15,63 +18,86 @@ export class ArticleDetail extends Component {
...
+            />
+            <br />
+            <br />
+            <div className='card bg-light text-dark'>
+              <div className='card-body'>
+                <div className='text-center'>
+                  {data.images === '' ? (
+                    <img
+                      src='https://www.maketecheasier.com/assets/uploads/2018/02/mac-no-camera-feature-image.png'
+                      alt=''
```
Comment (SelaDanti) [inline src/pages/ArticleDetail/index.jsx]: Don't leave alt attribute empty

**Review by MaggieChege — APPROVED** [review]
Looks Good to me

**Comment by kwameasiago** [comment]
Look good to me

**Review by cosmas28 — CHANGES_REQUESTED** [review]
The test coverage in the `ArticleDetail` component has reduced because of the changes you added, please test the lines you added.

File: src/store/modules/deleteArticle/index.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import { toast } from 'react-toastify';
+import {
+  ARTICLE_DELETE_REQUEST,
```
Comment (KMaina) [inline src/store/modules/deleteArticle/index.js]: change to `DELETE_ARTICLE_REQUEST`

File: src/store/modules/deleteArticle/index.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+});
+
+export const deleteArticleFailed = response => ({
+  type: ARTICLE_DELETE_FAILED,
+  isLoading: false,
+  isDeleted: false,
+  response
+});
+
+export const deletePostedArticle = (articleSlug, deletedStatus) => dispatch => {
```
Comment (KMaina) [inline src/store/modules/deleteArticle/index.js]: change to `deleteArticle`

File: src/store/modules/deleteArticle/index.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      toast.success('Article deleted Successfully');
+      dispatch(deleteArticleSuccess(res));
+    })
+    .catch(errors => {
+      toast.error(errors.response.data.detail);
+      dispatch(deleteArticleFailed(errors.response.data.detail));
+    });
+};
+
+export const deleteArticle = (state = initialState, action) => {
```
Comment (KMaina) [inline src/store/modules/deleteArticle/index.js]: change to `deleteArticleReducer`

**Review by cosmas28 — APPROVED** [review]

**Pull request merged**